service/history/decisionChecker.go

```diff
@@ -581,6 +581,31 @@ func (v *decisionAttrValidator) validateStartChildExecutionAttributes(
 	return nil
 }
 
+func (v *decisionAttrValidator) validateTaskListName(tl *workflow.TaskList, defaultVal string) error {
```

Comment:
This method does more than just `validate` - it changes the passed in task list with the `defaultVal`.. Perhaps `validateAndSetTaskListName` ?

Reply:
this code changed a bit, this comment is no longer relevant